Source/PlasticSourceControl/Private/PlasticSourceControlShell.cpp

```diff
@@ -99,7 +99,22 @@ static bool _StartBackgroundPlasticShell(const FString& InPathToPlasticBinary, c
 	verify(FPlatformProcess::CreatePipe(ShellInputPipeRead, ShellInputPipeWrite, true));	// For writing commands to cm shell child process
 #endif
 
+#if PLATFORM_WINDOWS
 	ShellProcessHandle = FPlatformProcess::CreateProc(*InPathToPlasticBinary, *FullCommand, bLaunchDetached, bLaunchHidden, bLaunchReallyHidden, nullptr, 0, *InWorkingDirectory, ShellOutputPipeWrite, ShellInputPipeRead);
+#else
+	// Update working directory
+	char OriginalWorkingDirectory[PATH_MAX];
+	getcwd(OriginalWorkingDirectory, PATH_MAX);
+	char* NewWorkingDirectory = TCHAR_TO_ANSI(*InWorkingDirectory);
```

Comment:
This is dangerous as TCHAR_TO_ANSI() returns a pointer to a temporary that should be used immediately, not stored.
Can you please just directly call chdir(TCHAR_TO_ANSI(*InWorkingDirectory));

Thanks a lot!
Sébastien

Reply:
Sure thing!

Reply:
I've updated it.